src/librustc/dep_graph/edges.rs

```diff
@@ -104,63 +160,52 @@ impl DepGraphEdges {
     /// you are not in a task; what is bad is *writing* to tracked
     /// state (and leaking data that you read into a tracked task).
     pub fn read(&mut self, v: DepNode) {
-        if self.current_node().is_some() {
-            let source = self.make_node(v);
-            self.add_edge_from_current_node(|current| (source, current))
+        match self.task_stack.last_mut() {
```

Comment:
The `OpenTask::Anon` variant is missing from the match, so reads during anonymous tasks are just dropped. I should have made the match exhaustive indeed `:)`

I'm in the process of fixing this but there are some complications (especially around https://github.com/rust-lang/rust/blob/bd32b1ba0d2d51a7e8505c1d3e37d17d3ba12843/src/librustc/traits/fulfill.rs#L628).